### PR TITLE
Create dedicated interconnect module.

### PR DESCRIPTION
**Pull request by daniel-cit**
Creation of  dedicated interconnect module in `3-networks`.

TODOs:

- [x] add  commented call in all the environments for both private and restricted shared VPC after https://github.com/terraform-google-modules/terraform-example-foundation/pull/72 is merged.
- [x] update README.md file in the new module and in each environment to presente the new module as the recommended option for onprem connection.

Depends on:

- https://github.com/terraform-google-modules/terraform-example-foundation/pull/72
- https://github.com/terraform-google-modules/terraform-example-foundation/pull/71

### CODE REVIEW

File: 3-networks/modules/dedicated_interconnect/main.tf [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+locals {
+  interconnect_project_id = data.google_projects.interconnect_project.projects[0].project_id
+}
+
+data "google_projects" "interconnect_project" {
+  filter = "labels.application_name=prj-interconnect"
+}
+
+module "interconnect_attachment1_region1" {
```
Comment (bharathkkb) [inline 3-networks/modules/dedicated_interconnect/main.tf:25]: Not too familiar with this module but does it make sense to have any outputs for these?

File: 3-networks/modules/dedicated_interconnect/main.tf [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    peer_ip_address = var.peer_ip_address
+    peer_asn        = var.peer_asn
+  }
+}
+
+module "interconnect_attachment2_region1" {
+  source  = "terraform-google-modules/cloud-router/google//modules/interconnect_attachment"
+  version = "~> 0.2.0"
+
+  name    = "vl-${var.interconnect_location2_region1}-${var.vpc_name}-${var.default_region1}-cr1"
```
Comment (rjerrems) [inline 3-networks/modules/dedicated_interconnect/main.tf:51]: should this be cr2?

File: 3-networks/modules/dedicated_interconnect/main.tf [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    peer_ip_address = var.peer_ip_address
+    peer_asn        = var.peer_asn
+  }
+}
+
+module "interconnect_attachment2_region2" {
+  source  = "terraform-google-modules/cloud-router/google//modules/interconnect_attachment"
+  version = "~> 0.2.0"
+
+  name    = "vl-${var.interconnect_location2_region2}-${var.vpc_name}-${var.default_region2}-cr1"
```
Comment (rjerrems) [inline 3-networks/modules/dedicated_interconnect/main.tf:95]: should this also be cr2?

**Comment by rjerrems** [comment]
Overall looking good aside from a couple of resource names. Should we also have some docs about what needs to happen before the attachments are created or is that in the whitepaper?

File: 3-networks/modules/dedicated_interconnect/README.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
+<!-- BEGINNING OF PRE-COMMIT-TERRAFORM DOCS HOOK -->
```
Comment (rjerrems) [inline 3-networks/modules/dedicated_interconnect/README.md:1]: We should provide a brief overview of what the module is designed to do, even if only a few sentences.

File: 3-networks/modules/dedicated_interconnect/main.tf (hunk 1 above)
Comment (daniel-cit) [reply 3-networks/modules/dedicated_interconnect/main.tf:25]: adding to the module output the same outputs of the underlying module

File: 3-networks/modules/dedicated_interconnect/main.tf (hunk 2 above)
Comment (daniel-cit) [reply 3-networks/modules/dedicated_interconnect/main.tf]: fixed

File: 3-networks/modules/dedicated_interconnect/main.tf (hunk 3 above)
Comment (daniel-cit) [reply 3-networks/modules/dedicated_interconnect/main.tf]: fixed

File: 3-networks/modules/dedicated_interconnect/README.md (hunk 4 above)
Comment (daniel-cit) [reply 3-networks/modules/dedicated_interconnect/README.md:15]: Added an explanation and link to public documentation and also added commented example calls from the environments.

File: 3-networks/modules/dedicated_interconnect/outputs.tf [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+output "interconnect_attachment1_region1" {
+  value       = module.interconnect_attachment1_region1.attachment
+  description = "The created attachment"
```
Comment (rjerrems) [inline 3-networks/modules/dedicated_interconnect/outputs.tf:19]: nit: these should be more specific. I.e "The interconnect attachment 1 for region 1"

**Pull request merged**